plugins/mistral/modelgauge/suts/mistral_sut.py

```diff
@@ -73,17 +73,119 @@ def translate_response(self, request: MistralAIRequest, response: MistralAIRespo
         return SUTResponse(completions=completions)
 
 
-MISTRAL_API_KEY = InjectSecret(MistralAIAPIKey)
+class MistralAIResponseWithModerations(BaseModel):
+    """Mistral's ChatCompletionResponse object + moderation scores."""
 
-model_name = "ministral-8b"
-model_version = "2410"
-model_full_name = "ministral-8b-latest"  # Mistral's endpoint schema
-model_uid = f"mistralai-{model_name}-{model_version}"
-SUTS.register(MistralAISut, model_uid, model_full_name, model_version, MISTRAL_API_KEY)
+    response: ChatCompletionResponse  # Contains multiple completions.
+    moderations: dict[int, ClassificationResponse]  # Keys correspond to a choice's index field.
 
 
-model_name = "mistral-large"
-model_version = "2411"
-model_full_name = "mistral-large-latest"  # Mistral's endpoint schema
-model_uid = f"mistralai-{model_name}-{model_version}"
-SUTS.register(MistralAISut, model_uid, model_full_name, model_version, MISTRAL_API_KEY)
+@modelgauge_sut(capabilities=[AcceptsTextPrompt])
+class MistralAIModeratedSut(PromptResponseSUT):
+    """A MistralAI SUT hosted on MistralAI with safety moderation based on the following recipe:
+    https://colab.research.google.com/github/mistralai/cookbook/blob/main/mistral/moderation/system-level-guardrails.ipynb#scrollTo=OlnXFkLo5sKX
+    """
+
+    def __init__(
+        self,
+        uid: str,
+        model_name: str,
+        moderation_model_name: str,
+        num_generations: int,
+        temperature: float,
+        threshold: float,
+        api_key: MistralAIAPIKey,
+    ):
+        assert (
+            num_generations > 1
+        ), "The moderation strategy uses a sampling-based mechanism. num_generations should be greater than 1."
+        super().__init__(uid)
+        self.model_name = model_name
+        self.moderation_model_name = moderation_model_name
+        self.num_generations = num_generations
+        self.temperature = temperature
+        self.threshold = threshold
+        self._api_key = api_key
+        self._client = None
+
+    @property
+    def client(self):
+        if not self._client:
+            self._client = MistralAIClient(self.model_name, self._api_key)
+        return self._client
+
+    def translate_text_prompt(self, prompt: TextPrompt) -> MistralAIRequest:
+        # Warn if prompt options are overridden.
+        if prompt.options.temperature is not None and prompt.options.temperature != self.temperature:
```

Comment:
Would this be clearer and have the same functionality if you did something like:

`if prompt.options.temperature and ...`?

Same for ones below?

Reply:
No because `0.0` evaluates to `False`!